src/gui/masterkey/KeyFileEditWidget.cpp

```diff
@@ -120,6 +122,25 @@ void KeyFileEditWidget::browseKeyFile()
     QString filters = QString("%1 (*.key);;%2 (*)").arg(tr("Key files"), tr("All files"));
     QString fileName = fileDialog()->getOpenFileName(this, tr("Select a key file"), QString(), filters);
 
+    if (QFileInfo(fileName).canonicalFilePath() == m_parent->getDatabase()->canonicalFilePath()) {
+        MessageBox::critical(getMainWindow(),
+                             tr("Invalid Key File"),
+                             tr("You cannot use the current database as its own keyfile. Please choose a different "
+                                "file or generate a new key file."));
+        return;
+    } else if (fileName.endsWith(".kdbx", Qt::CaseInsensitive)) {
+        auto response =
+            MessageBox::warning(getMainWindow(),
+                                tr("Invalid Key File"),
+                                tr("You have chosen a database file as your key file. This file will most likely "
+                                   "change and prevent you from unlocking your database. Do you want to continue?"),
+                                MessageBox::Continue | MessageBox::Cancel,
+                                MessageBox::Cancel);
```

Comment:
Please rephrase to something a bit more concise, such as:

Title: "Suspicious key file"
Message: "The chosen key file looks like a password database file.
A key file must be a static file that never changes or you will lose access to your database forever.
Are you sure you want to continue with this file?"

Reply:
Changes made